### PR TITLE
Passing endpoint, access and secret key to logical-backup container

### PR DESCRIPTION
**Pull request by ermajn**
logical-backup container needs to upload dump results to S3 storage, and when using non-AWS custom s3, it is nice to have endpoint variable arround. 
Same goes for access and secret key.

### CODE REVIEW

**Comment by FxKu** [comment]
Related to #592, but does not cover the `sse` option.

Could you also reflect these changes in the [logical backup script](https://github.com/zalando/postgres-operator/blob/master/docker/logical-backup/dump.sh)?

**Comment by sdudoladov** [comment]
@ermajn [ AWS CLI docs](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) say the following:

> Precedence of options

> If you specify an option by using one of the environment variables described in this topic, it overrides any value loaded from a profile in the configuration file.

Isn't it the case  that setting `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` to the empty string by default may override the existing conf and thus make backups unusable ?

**Comment by ermajn** [comment]
> @ermajn [ AWS CLI docs](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) say the following:
> 
> > Precedence of options
> 
> > If you specify an option by using one of the environment variables described in this topic, it overrides any value loaded from a profile in the configuration file.
> 
> Isn't it the case that setting `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` to the empty string by default may override the existing conf and thus make backups unusable ?

Probablly yes, acording to the documentation, you are right.
Than, default and empty string should not set environment variables. Does that seems ok for you?

**Comment by muff1nman** [comment]
Any way to help get this PR merged?

**Comment by ermajn** [comment]
> Any way to help get this PR merged?

As far as I can see, it is market to be in v1.3.
If I can do anything to merge it before, let me know.

**Comment by muff1nman** [comment]
The one awkward aspect of this is that this is storing secrets in a configmap, rather than a secret. In addition, the secret then gets propagated to any other namespace in the form of env values on the cronjobs. Not pretty.

Spitballing here but would it instead make sense to move the logical backup configuration to the postgresqls.acid.zalan.do namespaced objects? The idea would be that you could create a secret in the namespace for holding the access/secret keys and reference it from the postgresqls.acid.zalan.do. This would be a significant departure, but it might make more sense.

**Comment by ermajn** [comment]
> The one awkward aspect of this is that this is storing aws creds in a configmap, rather than in a secret. In addition, the secret then gets propagated to any other namespace in the form of env values on the cronjobs. Not pretty.
> 
> Spitballing here but would it instead make sense to move the logical backup configuration to the postgresqls.acid.zalan.do namespaced objects? The idea would be that you could create a secret in the namespace for holding the access/secret keys and reference it from the postgresqls.acid.zalan.do. This would be a significant change, but it might make more sense.

This pretty much make sense to me. 
Only in that case, reference would look up in secrets in namespace where other postgresqls.acid.zalan.do namespaced objects objects are, that means, no single secret can be used as it is happening now if credentials are stored in configmap.

**Comment by FxKu** [comment]
@ermajn could you add the parameters also to the default CRD-based [configuration file](https://github.com/zalando/postgres-operator/blob/master/manifests/postgresql-operator-default-configuration.yaml)? Use the same order as in my latest commit for the configMap. Please, inherit the alphabetical order also in the two values files of the helm chart. Makes finding stuff a bit easier in the end. Thanks :)

File: pkg/cluster/k8sres.go [hunk 1]
```diff
@@ -1571,6 +1571,14 @@ func (c *Cluster) generateLogicalBackupPodEnvVars() []v1.EnvVar {
 			Name:  "LOGICAL_BACKUP_S3_BUCKET",
 			Value: c.OpConfig.LogicalBackup.LogicalBackupS3Bucket,
 		},
+		{
+			Name:  "LOGICAL_BACKUP_S3_ENDPOINT",
+			Value: c.OpConfig.LogicalBackup.LogicalBackupS3Endpoint,
+		},
+		{
+			Name:  "AWS_SSE",
```
Comment (FxKu) [inline pkg/cluster/k8sres.go]: any reason why this env variable doesn't have the `LOGICAL_BACKUP_` prefix?

Comment (ermajn) [reply pkg/cluster/k8sres.go]: no, that was my mistake. Dump.sh is already expecting LOGICAL_BACKUP_S3_SSE. I've fixed that.

**Comment by FxKu** [comment]
:+1:

**Comment by sdudoladov** [comment]
:+1:

**Comment by sdudoladov** [comment]
@ermajn thank you for the contribution :)

**Comment by vitobotta** [comment]
@ermajn Hi! I'm using Scaleway object storage and uploads of the logical backups fail unless I pass the "--region" argument to the aws command. Not sure if this is only required with some storage services but it is definitely required with Scaleway. Would it be possible for you to do a new PR to add the setting for the region? I could try myself but I wouldn't know where to start and it would take a while for me to understand how to do it :) Thanks!

**Comment by vitobotta** [comment]
#812

**Comment by vitobotta** [comment]
Actually it wasn't difficult to replicate this PR for the region setting, so I created a new PR for that.

**Pull request merged**